src/main-process/atom-application.coffee

```diff
@@ -35,9 +36,14 @@ class AtomApplication
     unless options.socketPath?
       if process.platform is 'win32'
         userNameSafe = new Buffer(process.env.USERNAME).toString('base64')
```

Comment:
Let's join the USERNAME and ATOM_HOME together with a | separator and then MD5 and base64 it.  Will keep things quite unique while keeping the socket name length manageable.